core/common/src/main/java/alluxio/metrics/Metric.java

```diff
@@ -40,13 +40,14 @@
   private final MetricsSystem.InstanceType mInstanceType;
   private final String mHostname;
   private final String mName;
-  private final Double mValue;
   private final MetricType mMetricType;
   private String mInstanceId;
   // TODO(yupeng): consider a dedicated data structure for tag, when more functionality are added to
   // tags in the future
   private final Map<String, String> mTags;
 
+  private Double mValue;
```

Comment:
shouldn't this be an atomic boolean or a `volatile` variable for thread safety?

Reply:
Actually, `volatile` won't work since there is an increment operation. It also doesn't seem like AtomicDouble exists in java, but there is one in guava.